pkg/pf/tests/provider_checkconfig_test.go

```diff
@@ -802,3 +807,75 @@ func makeProviderServer(
 	require.NoError(t, err)
 	return server
 }
+
+func TestExtraConfig(t *testing.T) {
+	t.Parallel()
+
+	provBuilder := providerbuilder.NewProvider(
+		providerbuilder.NewProviderArgs{
+			AllResources: []providerbuilder.Resource{
+				providerbuilder.NewResource(providerbuilder.NewResourceArgs{
+					ResourceSchema: rschema.Schema{
+						Attributes: map[string]rschema.Attribute{
+							"s": rschema.StringAttribute{Optional: true},
+						},
+					},
+				}),
+			},
+			ProviderSchema: schema.Schema{
+				Attributes: map[string]schema.Attribute{
+					"config": schema.StringAttribute{Optional: true},
+				},
+			},
+		})
+
+	prov := bridgedProvider(provBuilder)
+
+	prov.ExtraConfig = map[string]*info.Config{
+		"extra": {
```

Comment:
Is this a TF or Pulumi name?

What happens with `extra_config` or `extraConfig`?

Reply:
It's a pulumi name AFAIK as it's in ProviderInfo struct

Reply:
I don't think that's relevant to the test though - the name transformations work the same everywhere.

Reply:
Can you extend the test to show how it interacts with the name transformation?